File: mario.c
#include <cs50.h>
#include <stdio.h>

int get_positive_int(string prompt);
void print_space(int num);
void print_hash(int num);

int main(void)
{
    int height = get_positive_int("Positive integer:");
    printf("%i\n", height);
    
    int spaces = height - 1;
    int hashes = 1;
    
    // Build pyramid
    for (int i = 0; i < height; i++)
    {
        print_space(spaces);
        print_hash(hashes);
        printf(" ");
        print_hash(hashes);
        printf("\n");
        spaces--;
        hashes++;
    }
}

// Prompt user for positive integer
int get_positive_int(string prompt)
{
    int n;
    do
    {
        n = get_int("Height:");
    }
    while (n < 1 || n > 8);
    return n;
}

// Print hashes
void print_hash(int num)
{
    for(int i = 0; i < num; i++)
    {
        printf("#");
    }
}

// Print spaces
void print_space(int num)
{
    for(int i = 0; i < num; i++)
    {
        printf(" ");
    }
}

File: mario.py
def print_hash(x):
    for i in range(1, x+1):
        print("#", end="")


def print_space(y):
    for i in range(1, y+1):
        print(" ", end="")


def get_positive_int():
    i = 0
    while i < 1 or i > 8:
        i = int(input("Height: "))
    else:
        return i


height = get_positive_int()
spaces = height - 1
hashes = 1

for x in range(1, height+1):
    print_space(spaces)
    print_hash(hashes)
    print("  ", end="")
    print_hash(hashes)
    print("")
    spaces -= 1
    hashes += 1
